### PR TITLE
feat: Repr for classes

### LINKED ISSUE #556: [Feature] add __repr__ methods
For example, trying to debug why my `page.on('request', fn)` doesn't work, there's no useful information when printing request objects:

```py
from playwright.sync_api import sync_playwright


def print_request(request):
    print(request)


def run(playwright):
    webkit = playwright.webkit
    browser = webkit.launch()
    context = browser.new_context()
    page = context.new_page()
    page.on("request", print_request)
    page.goto("https://github.com")
    browser.close()


with sync_playwright() as playwright:
    run(playwright)
```

```
<playwright._impl._network.Request object at 0x7f6f26943880>
<playwright._impl._network.Request object at 0x7f6f268d8a30>
<playwright._impl._network.Request object at 0x7f6f268d8af0>
<playwright._impl._network.Request object at 0x7f6f268d8bb0>
<playwright._impl._network.Request object at 0x7f6f268d8dc0>
<playwright._impl._network.Request object at 0x7f6f268d8ee0>
<playwright._impl._network.Request object at 0x7f6f267a9550>
```

### PR DESCRIPTION
**Pull request by kumaraditya303**
Closes #556

### CODE REVIEW

File: playwright/_impl/_browser.py [hunk 1]
```diff
@@ -150,6 +150,9 @@ async def close(self) -> None:
     def version(self) -> str:
         return self._initializer["version"]
 
+    def __repr__(self) -> str:
+        return f"<Browser type={self._browser_type} version={self.version} >"
```
Comment (mxschmitt) [inline playwright/_impl/_browser.py:154]:
```suggestion
        return f"<Browser type={self._browser_type} version={self.version}>"
```

File: playwright/_impl/_dialog.py [hunk 2]
```diff
@@ -41,3 +41,6 @@ async def accept(self, promptText: str = None) -> None:
 
     async def dismiss(self) -> None:
         await self._channel.send("dismiss")
+
+    def __repr__(self) -> str:
+        return f"<Dialog type={self.type} message={self.message} default={self.default_value} >"
```
Comment (mxschmitt) [inline playwright/_impl/_dialog.py:46]:
```suggestion
        return f"<Dialog type={self.type} message={self.message} default_value={self.default_value}>"
```

File: playwright/_impl/_browser_type.py [hunk 3]
```diff
@@ -130,6 +130,9 @@ async def launch_persistent_context(
                 raise not_installed_error(f'"{self.name}" browser was not found.')
             raise e
 
+    def __repr__(self) -> str:
+        return f"<BrowserType name={self.name} executable={self.executable_path}>"
```
Comment (mxschmitt) [inline playwright/_impl/_browser_type.py:134]:
```suggestion
        return f"<BrowserType name={self.name} executable_path={self.executable_path}>"
```

File: playwright/_impl/_download.py [hunk 4]
```diff
@@ -46,3 +46,6 @@ async def path(self) -> Optional[pathlib.Path]:
     async def save_as(self, path: Union[str, Path]) -> None:
         path = str(Path(path))
         return await self._channel.send("saveAs", dict(path=path))
+
+    def __repr__(self) -> str:
+        return f"<Download url={self.url!r} filename={self.suggested_filename!r}>"
```
Comment (mxschmitt) [inline playwright/_impl/_download.py:51]:
```suggestion
        return f"<Download url={self.url!r} suggested_filename={self.suggested_filename!r}>"
```

File: playwright/_impl/_file_chooser.py [hunk 5]
```diff
@@ -54,6 +54,9 @@ async def set_files(
     ) -> None:
         await self._element_handle.set_input_files(files, timeout, noWaitAfter)
 
+    def __repr__(self) -> str:
+        return f"<FileChooser page={self._page} element={self._element_handle} >"
```
Comment (mxschmitt) [inline playwright/_impl/_file_chooser.py:58]:
```suggestion
        return f"<FileChooser page={self._page} element={self._element_handle}>"
```

File: playwright/_impl/_transport.py [hunk 6]
```diff
@@ -96,3 +96,6 @@ def send(self, message: Dict) -> None:
         self._output.write(
             len(data).to_bytes(4, byteorder="little", signed=False) + data
         )
+
+    def __repr__(self) -> str:
+        return f"<Transport running={not self._stopped} executable={str(self._driver_executable)}>"
```
Comment (mxschmitt) [inline playwright/_impl/_transport.py:101]:
```suggestion
```
Its not exposed, so its not needed afaik.

**Review by mxschmitt — APPROVED** [review]
LGTM, tests would be great to have.

File: playwright/_impl/_transport.py (hunk 6 above)
Comment (kumaraditya303) [reply playwright/_impl/_transport.py:101]: Yeah, I was waiting for feedback so as to continue for tests

**Comment by kumaraditya303** [comment]
@mxschmitt Currently the class like `Browser` do not have `__repr__` as they are generated by script, the classes like `Route` are working fine as they refer to the `playwright._impl` package, there are some changes required to the script

**Comment by kumaraditya303** [comment]
@mxschmitt For tests, it requires some changes it api generation https://github.com/microsoft/playwright-python/pull/581/checks?check_run_id=2148192137#step:10:166

**Comment by mxschmitt** [comment]
> @mxschmitt For tests, it requires some changes it api generation https://github.com/microsoft/playwright-python/pull/581/checks?check_run_id=2148192137#step:10:166

Created a PR for you which unblocks it.

**Comment by kumaraditya303** [comment]
Thanks, @mxschmitt I have rebased on it and can close that PR

**Comment by mxschmitt** [comment]
> Thanks, @mxschmitt I have rebased on it and can close that PR

tests for the individual objects would be great to have, then we can merge it.

**Comment by kumaraditya303** [comment]
@mxschmitt Just added a bunch of tests

**Review by mxschmitt — APPROVED** [review]
LGTM, two more nits:
a) the Worker class `__repr__` method is not checked
b) would be cool if you could move the `__repr__` more up so its underneath the `__init__`
Thanks!

**Comment by kumaraditya303** [comment]
> LGTM, two more nits:
> a) the Worker class `__repr__` method is not checked
> b) would be cool if you could move the `__repr__` more up so its underneath the `__init__`
> Thanks! (I've checked the coverage using pytest-cov, to compare if actual all methods are covered)

Done!

**Review by mxschmitt — APPROVED** [review]

File: playwright/_impl/_frame.py [hunk 7]
```diff
@@ -81,6 +81,9 @@ def __init__(
             lambda params: self._on_frame_navigated(params),
         )
 
+    def __repr__(self) -> str:
+        return f"<Frame name={self.name} url={self.url!r}>"
```
Comment (pavelfeldman) [inline playwright/_impl/_frame.py:85]: Why !r? url is just a string...

File: scripts/generate_async_api.py [hunk 8]
```diff
@@ -48,6 +48,10 @@ def generate(t: Any) -> None:
     print("")
     print(f"    def __init__(self, obj: {class_name}Impl):")
     print("        super().__init__(obj)")
+    print("")
+    print("    def __repr__(self) -> str:")
```
Comment (pavelfeldman) [inline scripts/generate_async_api.py:52]: You should put it into ImplWrapper once instead of generating it everywhere.

**Review by pavelfeldman — COMMENTED** [review]
Great job, some notes inline!

File: playwright/_impl/_frame.py (hunk 7 above)
Comment (kumaraditya303) [reply playwright/_impl/_frame.py:85]: It would be in quotes and it looks better

File: scripts/generate_async_api.py (hunk 8 above)
Comment (kumaraditya303) [reply scripts/generate_async_api.py:52]: This wasn't done by me, @mxschmitt did it I rebased over it

Comment (mxschmitt) [reply scripts/generate_async_api.py:52]: @kumaraditya303 yes please add it to the `ImplWrapper` class instead. Thats better! and remove the generation logic which I've implemented.

Comment (kumaraditya303) [reply scripts/generate_async_api.py]: Ok Done ! @mxschmitt

**Review by mxschmitt — APPROVED** [review]

**Comment by kumaraditya303** [comment]
@mxschmitt can you review https://github.com/microsoft/playwright-pytest/pull/45

**Review by pavelfeldman — APPROVED** [review]

**Pull request merged**